BeagleBone_black/src/communications/communications.cpp

```diff
@@ -67,29 +69,42 @@ Communications::~Communications()
 int Communications::sendData(std::string message)
 {
   // Incoming strings should be terminated by "...\n".
-  memset(buffer, '\0', 256);
+  memset(buffer_, '\0', 256);
   const char *data = message.c_str();
-  n_ = write(sockfd_, data, message.length());  // ‘_size_t write(int, const void*, size_t)’
+  int n = write(sockfd_, data, message.length());  // ‘_size_t write(int, const void*, size_t)’
 
-  if (n_ < 0) {
+  if (n < 0) {
     log_.ERR("COMN", "CANNOT READ FROM SOCKET.\n");
   }
 
-  return atoi(buffer);
+  return atoi(buffer_);
+}
+
+int Communications::receiveRunLength()
+{
+  int n = read(sockfd_, buffer_, 255);
+  int run_length = atoi(buffer_);
+  log_.INFO("COMN", "Received track length of %f", static_cast<float>(run_length));
+
+  return run_length;
 }
 
 int Communications::receiveMessage()
 {
-  n_ = read(sockfd_, buffer, 255);
+  int n = read(sockfd_, buffer_, 255);
+  int command;
 
-  if (n_ < 0) {
+  if (n < 0) {
     log_.ERR("COMN", "CANNOT READ FROM SOCKET.\n");
-    command_ = 1;
+    command = 1;
```

Comment:
Could `return 1;` in this if block cos it doesn't make sense to log "Received 1 (STOP)" after this block i think? ;  could then move `int command = atoi(buffer);` to after this block;